### PR TITLE
Black Mesa polishing-of-ancient-content

### PR DESCRIPTION
**Pull request by Stalkeros2**
## About The Pull Request
Polishing, like, a truckload of old ideas that are sometimes pretty bad and adding new items that should've been added previously but I either went on a hiatus or forgot.
image
HECU's camp has been improved to be more roleplay-oriented. An actual canteen has been added on top of the armory, which, hopefully, will make people more inclined to roleplay and interact with each other.
The corridor before HECU has also been refined to make it more obvious that they're in a not-so-good condition. The addition of an infirmary will give Vanguard an additional breaching spot, and also assist HECU in their recovery or defense, as it can work as a potentially good defensive spot.
image

A lot of HECU equipment-related changes:
Ugly placeholder plate carrier/basic armor kit has been replaced with an armor that'll be equal to the Vanguard's stats-wise, yet, as usual, lacks any sort of NVGs. As a result of that change, HECU's bulletproof armor is now gone.
MP5 and Mesa-exclusive Glock 17 (now Glock 20)'s caliber changed from 9x21mm and 9x19mm Peacekeeper to 10mm Peacekeeper, to keep itcompatible with most of the assumed small-arms of the away mission.
Adds Mk-11.4 rifle as a cheaper alternative to M16.
Adds HE grenades and old stupidly powerful landmines to HECU vendor. Apparently, people who stepped on them (I am not even kidding) enjoyed that a tad too much.
Adds Vanguard's tomahawk to the melee option if someone suddenly decides to go melee, as well as some fairly limited (a single balaclava) clothing options to the miscellaneous options.
HECU now start with the weaker Bowie knives instead of the combat ones.
Also, HECU zombies.

You can now sell Xen crystals for 1,2k credits per piece; unaffected by the price decrease.
AND THE MOST IMPORTANT CHANGE,
Resonance Cascade is now gone. It was and is too bad to keep, and my idea of initial removal was also horribly terrible.

## How This Contributes To The Skyrat Roleplay Experience
Expanded HECU camp should hopefully increase their preparation time and, as a result, make them interact with each other more than before.
One of the main issues of HECU's area was the fact that even though it's expected to be at the very least visually run down (because they were trying to evacuate, yet there was really nothing to back that up), it actually was not. Hopefully, I've fixed that; the addition of an infirmary and HECU zombies is also meant to back that up and also give the marines an additional place to fortify and look through, thus, once again, increasing their preparation time.
Plate carrier was and is too ugly and I was asked a lot of times to remove it. After a while of trying to find a suitable replacement (and failing in the process), I believe that the new option should come as a better alternative.
MP5's caliber change is meant to make it both compatible with the Glock for immersion purposes.
M16 is usually reserved as the last option to buy due to HECU's now decreased points. And thus, the Mk-11.4 should come in handy as a economical alternative to it.
HE grenades are mostly flavor due to an almost similar functionality to the normal frag grenades. Landmines, as already mentioned, were requested by the people that got blown up, so I assume it is a good idea to keep them.
Tomahawk is simply fun to use in a place that's full of ranged weaponry; clothing is meant to give HECU some replacement outfits, as well as a very insignificant drip.

Resonance Cascade was simply too bad and unenjoyable to keep and, once again, I really should've removed it entirely in the previous PR, yet didn't anticipate my solution to be incredibly horrible.

## Changelog

:cl: Stalkeros, Lukaster42
add: HECU zombies for Black Mesa.
del: Resonance Cascade and related mechanics.
qol: HECU camp and nearby corridors' expansion.
balance: HECU's basic armor redesigned to be similar to the Vanguard's stats-wise.
balance: MP5's and Mesa Glock's caliber changed to 10mm Peacekeeper.
balance: Adds Mk-11.4 to HECU vendor, among other things.
/:cl:

### CODE REVIEW

**Comment by Stalkeros2** [comment]
> You currently have a negative Fix/Feature pull request delta of -14. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

image

File: modular_skyrat/modules/black_mesa/code/armaments/_armament_primary.dm [hunk 1]
```diff
@@ -66,7 +70,12 @@
 /obj/item/storage/box/ammo_box/shotgun_12g
 
 /obj/item/storage/box/ammo_box/shotgun_12g/PopulateContents()
-	var/funshell = pick(/obj/item/ammo_box/advanced/s12gauge/pt20, /obj/item/ammo_box/advanced/s12gauge/incendiary, /obj/item/ammo_box/advanced/s12gauge/flechette, /obj/item/ammo_box/advanced/s12gauge/beehive, /obj/item/ammo_box/advanced/s12gauge/iceblox, /obj/item/ammo_box/advanced/s12gauge/rip)
+	var/funshell = pick(/obj/item/ammo_box/advanced/s12gauge/pt20,
+						/obj/item/ammo_box/advanced/s12gauge/incendiary,
+						/obj/item/ammo_box/advanced/s12gauge/flechette,
+						/obj/item/ammo_box/advanced/s12gauge/beehive,
+						/obj/item/ammo_box/advanced/s12gauge/iceblox,
+						/obj/item/ammo_box/advanced/s12gauge/rip,)
```
Comment (Zonespace27) [inline modular_skyrat/modules/black_mesa/code/armaments/_armament_primary.dm]:
```suggestion
						/obj/item/ammo_box/advanced/s12gauge/rip,
						)
```
this should be a static list on the ammo box anyway

File: modular_skyrat/modules/black_mesa/code/armaments/armament_utility.dm [hunk 2]
```diff
@@ -70,3 +63,8 @@
 	item_type = /obj/item/radio/headset/headset_faction/bowman
 	max_purchase = 3
 	cost = 4
+
+/datum/armament_entry/hecu/utility/breaching_hammer ///Mesa more like R6S amirite
```
Comment (Zonespace27) [inline modular_skyrat/modules/black_mesa/code/armaments/armament_utility.dm]:
```suggestion
/datum/armament_entry/hecu/utility/breaching_hammer //Mesa more like R6S amirite
```

File: modular_skyrat/modules/black_mesa/code/xen_crystal.dm [hunk 3]
```diff
@@ -70,3 +64,12 @@
...
 	qdel(src)
+
+/datum/export/xen_crystal
+	cost = CARGO_CRATE_VALUE * 6 //1200
+	unit_name = "anomalous crystal sample"
+	export_types = list(/obj/item/grenade/xen_crystal)
+	include_subtypes = FALSE
+
+/datum/export/xen_crystal/sell_object(obj/O, datum/export_report/report, dry_run, apply_elastic = FALSE)
+	. = ..()
```
Comment (Zonespace27) [inline modular_skyrat/modules/black_mesa/code/xen_crystal.dm]: have you tested that this works

Comment (Zonespace27) [inline modular_skyrat/modules/black_mesa/code/xen_crystal.dm]:
```suggestion
	return ..()
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/black_mesa/code/mobs/nihilanth.dm [hunk 4]
```diff
@@ -66,28 +66,7 @@
 
 /mob/living/simple_animal/hostile/blackmesa/xen/nihilanth/death(gibbed)
 	. = ..()
```
Comment (ghost) [inline modular_skyrat/modules/black_mesa/code/mobs/nihilanth.dm:68]: All this proc does is call parent - does it need to be kept?

File: modular_skyrat/modules/black_mesa/code/objects.dm [hunk 5]
```diff
@@ -4,10 +4,10 @@
 	icon = 'modular_skyrat/modules/aesthetics/landmine/mines.dmi'
 	icon_state = "uglymine"
 
-	mine_type = /obj/effect/mine/explosive/mesa
+	mine_type = /obj/effect/mine/explosive/nri
 
 /// A nerfed down version of a landmine I really should've done a long time ago but only did it now.
-/obj/effect/mine/explosive/mesa
```
Comment (ghost) [inline modular_skyrat/modules/black_mesa/code/objects.dm:10]: dont understand why this path needs changing

File: modular_skyrat/modules/black_mesa/code/turfs.dm [hunk 6]
```diff
@@ -86,3 +86,26 @@
 		var/mob/living/unlucky_mob = arrived
 		unlucky_mob.adjustFireLoss(acid_damage)
 		playsound(unlucky_mob, 'sound/weapons/sear.ogg', 100, TRUE)
+
+/turf/open/water/electric
+	name = "electric water"
```
Comment (ghost) [inline modular_skyrat/modules/black_mesa/code/turfs.dm:91]: actually kind of based, love environmental hazards

File: modular_skyrat/modules/mapping/code/areas/away_content.dm [hunk 7]
```diff
@@ -162,61 +162,65 @@
...
 	icon_state = "awaycontent10"
 
+/area/awaymission/black_mesa/hecu_zone_infirmary
+	name = "Black Mesa HECU Territory Infirmary"
+	icon_state = "awaycontent11"
+
 /area/awaymission/black_mesa/hecu_zone_bathroom
 	name = "Black Mesa HECU Territory Bathroom"
-	icon_state = "awaycontent11"
+	icon_state = "awaycontent12"
```
Comment (ghost) [inline modular_skyrat/modules/mapping/code/areas/away_content.dm:171]: These icon states, I do not like them, no I do not

No clue if they're worth fixing though as I've never seen them used outside this mission

File: modular_skyrat/modules/black_mesa/code/mobs/nihilanth.dm (hunk 4 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/black_mesa/code/mobs/nihilanth.dm:68]: Yes, or else the Nihilanth doesn't get gibbed.

File: modular_skyrat/modules/black_mesa/code/objects.dm (hunk 5 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/black_mesa/code/objects.dm:10]: I uh.
Forgot to organise this properly.

**Comment by Stalkeros2** [comment]
Receiving mixed feedback about the mines; I'll probably come up with a solution later.

**Comment by Kush1Push1** [comment]
undocumented changes, why ipc can be hecu lorewise

**Comment by Lukaster42** [comment]
> undocumented changes, why ipc can be hecu lorewise

Have you ever heard about the german HL1 version? they replaced HECU with IPCs.
We decided to add them.

**Comment by Kush1Push1** [comment]
> > undocumented changes, why ipc can be hecu lorewise
> 
> Have you ever heard about the german HL1 version? they replaced HECU with IPCs. We decided to add them.

It doesn't fit well with, at least, my view about timeline and level of technologies during HL1. Marines with robots seems too futuristic.
And so I wouldn't roleplay it, tbh. And other people can have the same opinion. Any immersion ~~in gateway's team deathmatch~~ disappears.

**Comment by Lukaster42** [comment]
> > > undocumented changes, why ipc can be hecu lorewise
> > 
> > 
> > Have you ever heard about the german HL1 version? they replaced HECU with IPCs. We decided to add them.
> 
> It doesn't fit well with, at least, my view about timeline and level of technologies during HL1. Marines with robots seems too futuristic. And so I wouldn't roleplay it, tbh. And other people can have the same opinion. Any immersion ~in gateway's team deathmatch~ disappears.

-Lambada teleport.
-HEV
-Gluon Gun
-PCV
"Too futuristic."

File: modular_skyrat/modules/black_mesa/code/ghost_spawners.dm [hunk 8]
```diff
@@ -6,7 +6,9 @@
 	outfit = /datum/outfit/science_team
 	you_are_text = "You are a scientist in a top secret government facility. You blacked out. Now, you have woken up to the horrors that lay within."
 	flavour_text = "You are a scientist in a top secret government facility. You blacked out. Now, you have woken up to the horrors that lay within."
-	restricted_species = list(/datum/species/human)
+	important_text = "Do not try to explore the level unless Vanguard is already present and close to you, and even then, wait at least 20 minutes before leaving your area. \
+					Allowed races are humans and IPCs."
+	restricted_species = list(/datum/species/human, /datum/species/robotic/ipc)
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/black_mesa/code/ghost_spawners.dm]: nyet

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]
nyet no ipc hecu

File: modular_skyrat/modules/black_mesa/code/ghost_spawners.dm [hunk 9]
```diff
@@ -36,25 +38,26 @@
...
 	gloves = /obj/item/clothing/gloves/color/black
 	suit = /obj/item/clothing/suit/armor/vest/blueshirt
 	shoes = /obj/item/clothing/shoes/jackboots
+	belt = /obj/item/storage/belt/security/full
 	back = /obj/item/storage/backpack
-	backpack_contents = list(/obj/item/radio, /obj/item/gun/ballistic/automatic/pistol/g17/mesa, /obj/item/ammo_box/magazine/multi_sprite/g17)
+	backpack_contents = list(/obj/item/radio,
+							/obj/item/gun/ballistic/automatic/pistol/g17/mesa,
+							/obj/item/ammo_box/magazine/multi_sprite/ladon,
+							/obj/item/ammo_box/magazine/multi_sprite/ladon,)
```
Comment (Zonespace27) [inline modular_skyrat/modules/black_mesa/code/ghost_spawners.dm]:
```suggestion
							/obj/item/ammo_box/magazine/multi_sprite/ladon,
						)
```
and generally just even out all these list items

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by Gandalf2k15** [comment]
sex

**Comment by Stalkeros2** [comment]
Oh good god.

**Comment by Gandalf2k15** [comment]
fix conficks

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**